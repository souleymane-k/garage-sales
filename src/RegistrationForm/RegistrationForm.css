*{
    box-sizing: border-box;
  }
.signUpForm-form{
  position:relative;
}

.registrationForm{
  width:50%;
  height:500px;
}
.signUpForm-form{
  
  height:1000px;
}


input{
width:100%;
line-height: 2em;
padding:0.5em;
border-radius: 2px;
}
.btn{
width:100%;
line-height: 2em;
padding:0.5em;
border-radius: 2px;
background-color: red;
color: white;
font-size: large;
font-weight: 100px;
cursor:pointer;
}
@media all and (max-width: 640px) {

  .registrationForm{
    width:50%;
    height:500px;
  }
.btn{
  width:100%;
  line-height: 2em;
  padding:0.5em;
  border-radius: 2px;
  margin:0 auto;
  background-color: red;
  color:white;
  font-size: large;
  font-weight: 80px;
  cursor:pointer;
}
  }
