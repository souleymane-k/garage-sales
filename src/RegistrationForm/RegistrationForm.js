import React, {useState} from 'react';
import './RegistrationForm.css';
import AuthApiService from '../services/auth-api-service'
// import {Redirect} from 'react-router-dom'

export default function RegistrationForm (props) {

  // static defaultProps = {
  //   onLoginSuccess: () => {}
  // }

  // state = { error: null }
 const [error, setError] = useState(null);

  const handleSubmitJwtAuth = ev => {
    ev.preventDefault()
    setError(null);

    const { username, password, email } = ev.target

    AuthApiService.createUser({
      username: username.value,
      password: password.value,
      email:email.value,
    })
      .then(res => {
        username.value = ''
        password.value = ''
        props.onLoginSuccess()
      })
      .catch(res => {
        setError(res.error)
      })
  }
  // return <Redirect to='login'/>

  
    return (
      <form className='signUpForm-form' onSubmit={handleSubmitJwtAuth}>
        <fieldset className="registrationForm">
          <h3>Create an account</h3>
          <div className="form-group">
          <label className='signUpForm-form__username-label' htmlFor='signUpForm-form__username'>
            Username
          </label>
          <input
            type='text'
            name='username'
            id='SignUpForm-form__username'
            placeholder='username'
            required
          />
          </div>
          <div className="form-group">
          <label className='signUpForm-form__email-label' htmlFor='signUpForm-form__email'>
            Email
          </label>
          <input
            type='email'
            name='email'
            id='SignUpForm__email'
            placeholder='email'
            required
          />
          </div>
          <div className="form-group">
          <label className='signUpForm-form__password-label' htmlFor='signUpForm-form__password'>
            Password
          </label>
          <input
            type='password'
            name='password'
            id='SignUpForm-form__password'
            placeholder='password'
            required
          />
          </div>
          <div className="form-group">
          <label className='SignUpForn-form__password-repeat-label' htmlFor='signUpForm-form__password-repeat'>
            Confirm Password
          </label>
          <input
            type='password'
            name='passwordRepeat'
            id='SignUpForn-form__password-repeat'
            placeholder='confirm password'
            required
          />
          </div>
          <button className="btn" type='submit'>Submit</button>
          {error}
        </fieldset>
  
      </form>
    );
  
};

RegistrationForm.defaultProps = {
  onLoginSuccess: () => {}
}
