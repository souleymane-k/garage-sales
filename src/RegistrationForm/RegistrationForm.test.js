import React from 'react';
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import RegistrationForm from './RegistrationForm';

describe(`RegistrationForm component`, () => {
  it('renders the complete form', () => {
    const wrapper = shallow(<RegistrationForm />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<RegistrationForm />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
})
