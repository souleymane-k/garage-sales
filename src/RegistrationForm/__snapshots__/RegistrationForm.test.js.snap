// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistrationForm component renders the complete form 1`] = `
<form
  className="signUpForm-form"
  onSubmit={[Function]}
>
  <fieldset
    className="registrationForm"
  >
    <h3>
      Create an account
    </h3>
    <div
      className="form-group"
    >
      <label
        className="signUpForm-form__username-label"
        htmlFor="signUpForm-form__username"
      >
        Username
      </label>
      <input
        id="SignUpForm-form__username"
        name="username"
        placeholder="username"
        required={true}
        type="text"
      />
    </div>
    <div
      className="form-group"
    >
      <label
        className="signUpForm-form__email-label"
        htmlFor="signUpForm-form__email"
      >
        Email
      </label>
      <input
        id="SignUpForm__email"
        name="email"
        placeholder="email"
        required={true}
        type="email"
      />
    </div>
    <div
      className="form-group"
    >
      <label
        className="signUpForm-form__password-label"
        htmlFor="signUpForm-form__password"
      >
        Password
      </label>
      <input
        id="SignUpForm-form__password"
        name="password"
        placeholder="password"
        required={true}
        type="password"
      />
    </div>
    <div
      className="form-group"
    >
      <label
        className="SignUpForn-form__password-repeat-label"
        htmlFor="signUpForm-form__password-repeat"
      >
        Confirm Password
      </label>
      <input
        id="SignUpForn-form__password-repeat"
        name="passwordRepeat"
        placeholder="confirm password"
        required={true}
        type="password"
      />
    </div>
    <button
      className="btn"
      type="submit"
    >
      Submit
    </button>
  </fieldset>
</form>
`;
